e2e/deleteapi_test.go

```diff
@@ -16,12 +19,22 @@ func TestDeleteAPI(t *testing.T) {
 
 	// delete any api-server pods
 	deletedPods := make(map[string]struct{})
-	for _, pod := range apiPods.Items {
-		if err := client.CoreV1().Pods("kube-system").Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{}); err != nil {
-			t.Fatalf("error deleting api-server pod: %v", err)
+	wait.Poll(5*time.Second, 10*time.Minute, func() (bool, error) {
```

Comment:
10 minutes seems really long to wait for the API servers to be deleted.

Reply:
Also, don't we need to error if wait.Poll never succeeds?

Reply:
decreased to 2 minutes in c6e8a3d

added a return to retry the delete